File: app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/ui/SplashActivity.java
package com.berkatfaatulohalawa1711010164.facevoting.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.berkatfaatulohalawa1711010164.facevoting.API.APIService;
import com.berkatfaatulohalawa1711010164.facevoting.API.NoConnectivityException;
import com.berkatfaatulohalawa1711010164.facevoting.MainActivity;
import com.berkatfaatulohalawa1711010164.facevoting.R;
import com.berkatfaatulohalawa1711010164.facevoting.config.Constants;
import com.berkatfaatulohalawa1711010164.facevoting.helper.MyFirebaseMessagingService;
import com.berkatfaatulohalawa1711010164.facevoting.helper.SessionHelper;
import com.berkatfaatulohalawa1711010164.facevoting.model.LoginModel;
import com.berkatfaatulohalawa1711010164.facevoting.response.GetMenu;
import com.google.firebase.iid.FirebaseInstanceId;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences(
                Constants.USER_KEY, Context.MODE_PRIVATE);

        if(SessionHelper.sudahLogin(this)){



            String idUser = sharedPreferences.getString("id_user", "");
            get_status(idUser);
            String valds = sharedPreferences.getString("validasi", "");
            if(valds.equals("2")){
                Intent intent = new Intent(SplashActivity.this, Checkpoint.class);
                startActivity(intent);
                finish();
            }else{
                Intent intent = new Intent(SplashActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        }

        TextView btnStart = findViewById(R.id.tombol_start);

        btnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(SplashActivity.this, preLogin1.class);
                startActivity(intent);
            }
        });
    }

    private void get_status(String idUser)
    {
        try{
            Call<LoginModel> call = APIService.Factory.create(getApplicationContext()).dapatstatus(idUser);
            call.enqueue(new Callback<LoginModel>() {
                @Override
                public void onResponse(Call<LoginModel> call, Response<LoginModel> response) {
                    if(response.isSuccessful()) {
                        if (response.body() != null) {
                            SessionHelper.login(SplashActivity.this, response.body().getIdUser(), response.body().getToken_login(),response.body().getValidasi(),response.body().getNama(),response.body().getIdentitas());
                        }
                    }
                }

                @Override
                public void onFailure(Call<LoginModel> call, Throwable t) {
                    if(t instanceof NoConnectivityException) {
                        displayExceptionMessage("Offline, cek koneksi internet anda!");
                    }
                }
            });
        }catch (Exception e){
            e.printStackTrace();
            displayExceptionMessage(e.getMessage());
        }

    }

    @Override
    public void onResume(){
        super.onResume();
        SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences(
                Constants.USER_KEY, Context.MODE_PRIVATE);
        String valds = sharedPreferences.getString("validasi", "");
        if(SessionHelper.sudahLogin(this)){
            if(valds.equals("2")){
                Intent intent = new Intent(SplashActivity.this, Checkpoint.class);
                startActivity(intent);
                finish();
            }else{
                Intent intent = new Intent(SplashActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        }
    }

    public void displayExceptionMessage(String msg)
    {
        Toast.makeText(getApplicationContext(), msg, Toast.LENGTH_LONG).show();
    }
}

File: android/facevoting/app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/ui/SplashActivity.kt
package com.berkatfaatulohalawa1711010164.facevoting.ui

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import com.berkatfaatulohalawa1711010164.facevoting.MainActivity
import com.berkatfaatulohalawa1711010164.facevoting.core.Constants
import com.berkatfaatulohalawa1711010164.facevoting.core.Resource
import com.berkatfaatulohalawa1711010164.facevoting.core.SessionHelper
import com.berkatfaatulohalawa1711010164.facevoting.viewmodel.AuthViewModel

@SuppressLint("CustomSplashScreen")
class SplashActivity : AppCompatActivity() {
    private val viewModel: AuthViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        installSplashScreen()
        super.onCreate(savedInstanceState)
        val prefs = applicationContext.getSharedPreferences(Constants.USER_KEY, MODE_PRIVATE)
        if (SessionHelper.sudahLogin(this)) {
            val idUser = prefs.getString("id_user", "") ?: ""
            viewModel.refreshStatus(idUser)
            viewModel.refreshStatusState.observe(this) { resource ->
                if (resource is Resource.Success) {
                    val body = resource.data
                    SessionHelper.login(this, body.idUser, body.token_login, body.validasi, body.nama, body.identitas)
                } else if (resource is Resource.Error) {
                    Toast.makeText(applicationContext, resource.message, Toast.LENGTH_LONG).show()
                }
            }
            val valds = prefs.getString("validasi", "")
            if (valds == "2") {
                startActivity(Intent(this, Checkpoint::class.java))
            } else {
                startActivity(Intent(this, MainActivity::class.java))
            }
        } else {
            startActivity(Intent(this, WelcomeActivity::class.java))
        }
        finish()
    }
}
